fsw/src/bpnode_eds_dispatch.c

```diff
@@ -0,0 +1,87 @@
+/************************************************************************
+ * NASA Docket No. GSC-18,719-1, and identified as “core Flight System: Bootes”
+ *
+ * Copyright (c) 2020 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ************************************************************************/
+
+/**
+ * \file
+ *   This file contains the source code for the BPNode EDS bindings
+ */
+
+/*
+** Include Files:
+*/
+#include "bpnode_app.h"
+#include "bpnode_dispatch.h"
+#include "bpnode_cmds.h"
+#include "bpnode_eventids.h"
+#include "bpnode_msgids.h"
+#include "bpnode_msg.h"
+
+#include "bpnode_eds_dispatcher.h"
+#include "bpnode_eds_dictionary.h"
+
+/*
+ * Define a lookup table for BPNode command codes
+ */
+static const EdsDispatchTable_BPNode_Application_CFE_SB_Telecommand_t BPNODE_TC_DISPATCH_TABLE = {
+    .CMD     = {.NoopCmd_indication          = BPNode_NoopCmd,
+                .ResetCountersCmd_indication = BPNode_ResetCountersCmd
+               },
+    .SEND_HK = {.indication = BPNode_SendHkCmd}
+};
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * **/
+/*                                                                            */
+/*  Purpose:                                                                  */
+/*     This routine will process any packet that is received on the BPNode    */
+/*     command pipe.                                                          */
+/*                                                                            */
+/* * * * * * * * * * * * * * * * * * * * * * * *  * * * * * * *  * *  * * * * */
+void BPNode_TaskPipe(const CFE_SB_Buffer_t *SBBufPtr)
```

Comment:
Where are there two definitions of this function (`void BPNode_TaskPipe(const CFE_SB_Buffer_t *SBBufPtr)`)?

- First here (in `fsw/src/bpnode_eds_dispatch.c`)
- Second in `fsw/src/bpnode_dispatch.c`

I might need a walk-through of the intent behind this.

Reply:
Oh, I see now that our `CMakeLists.txt` checks for `CFE_EDS_ENABLED_BUILD` to be set, and switches between these two implementations. I have a few questions.

- Is this a new cFS standard?
- Is there any way to identify at runtime whether the app was built with this flag set?

If the answers aren't simple, perhaps we could pull the team together to discuss this and make sure we're all on the same page.

Reply:
This is the new cFS standard. I don't know that there's a way at runtime to distinguish, just at compile time

Reply:
Maybe we could add this to the info event we generate on the no-op (alongside the version info)?

Reply:
Is it useful information for operators to have? We can get that info from the build output and I feel like beyond testing the two versions of the build, it shouldn't matter

Reply:
I've created two issues to follow-up on this. First, on the DTN side, to update our BPNode later:

- https://etdjira.gsfc.nasa.gov/browse/DTNN-714

Next, on the cFS side, to recommend this be a standard for cFS apps:

- https://etdjira.gsfc.nasa.gov/browse/CORE-35

Good 'nuf for now :)